acceptancetests/assess_storage.py

```diff
@@ -312,7 +348,17 @@ def assess_storage(client, charm_series):
     expected = make_expected_ls(client, 'data/4', 'dummy-storage-np/0')
     check_storage_list(client, expected)
     log.info('Filesystem tmpfs PASSED')
+
     client.remove_service('dummy-storage-np')
+    # data/4 is a persistent storage, detach is required before removal.
+    wait_for_storage_detach(
+        client, storage_id='data/4', interval=15, timeout=60)
+    # data/4 is a persistent storage, need to be removed, otherwise it will
+    # interfere the next test result.
+    client.get_juju_output(
```

Comment:
FYI, you can run `juju remove-storage --force data/4`, obviating the need to wait for the detach to complete.